adapter-manager.js

```diff
@@ -145,24 +145,97 @@ class AdapterManager extends EventEmitter {
   }
 
   /**
+   * @method getThings
+   * @returns Returns an dictionary of all of the known things.
+   *          The dictionary key corresponds to the device id.
+   */
+  getThings() {
+    var things = [];
+    for (var thingId in this.devices) {
+      things.push(this.getThing(thingId));
+    }
+    return things;
+  }
+
+  /**
+   * @method getThing
+   * @returns Returns the thing with the indicated id.
+   */
+  getThing(thingId) {
+    var device = this.getDevice(thingId);
+    if (device) {
+      return device.getThing();
+    }
+  }
+
+  /**
+   * @method getProperties
+   * @returns Retrieves all of the properties associated with the thing
+   *          identified by `thingId`.
+   */
+  getProperties(thingId) {
+    var device = this.getDevice(thingId);
+    if (device) {
+      return device.getProperties();
+    }
+  }
+
+  /**
+   * @method getProperty
+   * @returns Retrieves the property named `propertyName` from the thing
+   *          identified by `thingId`.
+   */
+  getProperty(thingId, propertyName) {
+    var device = this.getDevice(thingId);
+    if (device) {
+      return device.getProperty(propertyName);
+    }
+  }
+
+  /**
+   * @method getPropertyValue
+   * @returns Retrieves the value of the property named `propertyName` from
+   *          the thing identified by `thingId`.
+   */
```

Comment:
ok. so perhaps what I've called getProperty should really be called getPropertyDescription?
And getPropertyValue should be called getProperty?

I don't have a problem with having getPropertyValue return JSON instead of a plain return value (too much C background on my part :), and it probably makes sense anyways if there was something more complex than a simple value.